test/e2e/atlas/helper_test.go

```diff
@@ -737,6 +740,92 @@ func createDBUserWithCert(projectID, username string) error {
 	return nil
 }
 
+func createDataFederationForProject(projectID string) (string, error) {
+	n, err := e2e.RandInt(1000)
+	if err != nil {
+		return "", err
+	}
+	dataFederationName := fmt.Sprintf("e2e-data-federation-%v", n)
+
+	opts := mongodbatlas.DataFederationInstance{
+		DataProcessRegion: &mongodbatlas.DataProcessRegion{
+			CloudProvider: "AWS",
+			Region:        "DUBLIN_IRL",
+		},
+		Storage: &mongodbatlas.DataFederationStorage{
+			Databases: []*mongodbatlas.DataFederationDatabase{
+				{
+					MaxWildcardCollections: 100,
+					Name:                   "testName",
+					Collections: []*mongodbatlas.DataFederationCollection{
+						{
+							DataSources: []*mongodbatlas.DataFederationDataSource{
+								{
+									Path:      "/airbnb/listingsAndReviews/{bedrooms int}/{review_scores.review_scores_rating int}/",
+									StoreName: "sample-data-atlas-data-lake",
+								},
+							},
+							Name: "testName",
+						},
+					},
+				},
+			},
+			Stores: []*mongodbatlas.DataFederationStore{
+				{
+					Name:      "sample-data-atlas-data-lake",
+					Provider:  "s3",
+					Bucket:    "sample-data-atlas-data-lake",
+					Delimiter: "/",
+					Public:    pointer.Get(true),
+					Region:    "US_EAST_1",
+				},
+			},
+		},
+		Name: dataFederationName,
+	}
+	publicKey, privateKey := os.Getenv("MCLI_PUBLIC_API_KEY"), os.Getenv("MCLI_PRIVATE_API_KEY")
```

Comment:
this could break in the future, these helpers rely on the CLI since the CLI will abstract the env var part and in the future once we remove mongocli from this repo we will stop supporting these envs, if you guys will add this you guys are in the hook for the support latter on cc @igor-karpukhin

Reply:
also, and this will be a hard line, use the new SDK if we are keeping this as it is

Reply:
yes

Reply:
We may be able to avoid creating our own client to create a dataFed (as we are now) by using `atlas dataFederation create` command. This method was avoided since the command was thought to be non-functional. If flag `--file` is used, a json file with the desired specs for testing should work. 
Alternatively, the env var could be bypassed to allow for the rest of the code to remain as is. I'm unsure how this could be done so I will have to defer to @igor-karpukhin @helderjs

Reply:
Got some hints from Igor on this, will try to fix in the next update.

Reply:
I saw a green in evergreen. Closing for now.

Reply:
Sent a possible fix. Not sure is 100% right as the original had more information that I could not make pass through the CLI interface for create data federated name. We'll see in the tests.

Reply:
will comment on specific issues now